.github/workflows/main.yml

```diff
@@ -57,112 +57,79 @@ jobs:
           python setup.py sdist --dist-dir ./wheels/dev/
           find ./wheels/dev -type f
 
-  build-test-env-base:
+  test-wheels:
     runs-on: ubuntu-latest
+    container:
+      image: opendatacube/datacube-statistician:latest
 
-    steps:
-      - uses: actions/checkout@v2
-
-      - uses: actions/cache@v2
-        id: conda_cache
-        with:
-          path: /tmp/test_env
-          key: ${{ runner.os }}-test-env-py38-\
-            ${{ hashFiles('tests/test-env-py38.yml') }}
-
-      - uses: conda-incubator/setup-miniconda@v2
-        if: steps.conda_cache.outputs.cache-hit != 'true'
-        with:
-          channels: conda-forge,defaults
-          channel-priority: true
-          activate-environment: ""
-          mamba-version: "*"
-          use-mamba: true
-
-      - name: Dump Conda Environment Info
-        shell: bash -l {0}
-        if: steps.conda_cache.outputs.cache-hit != 'true'
-        run: |
-            conda info
-            conda list
-            mamba -V
-            conda config --show-sources
-            conda config --show
-            printenv | sort
-
-      - name: Build Python Environment for Testing
-        shell: bash -l {0}
-        if: steps.conda_cache.outputs.cache-hit != 'true'
-        run: |
-          mamba env create -f tests/test-env-py38.yml -p /tmp/test_env
-
-      - name: Check Python Env
-        shell: bash -l {0}
-        if: steps.conda_cache.outputs.cache-hit != 'true'
-        run: |
-          mamba env export -p tests/env
-
-
-  test-with-coverage:
-    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgis/postgis:10-2.5
+        env:
+          POSTGRES_DB: opendatacube
+          POSTGRES_PASSWORD: opendatacubepassword
+          POSTGRES_USER: opendatacube
 
     needs:
-      - build-test-env-base
+      - build-wheels
 
     steps:
       - uses: actions/checkout@v2
-      - uses: rickstaa/action-black@v1
-        with:
-          black_args: "--check odc/stats"
-
-      - name: Get Conda Environment from Cache
-        uses: actions/cache@v2
-        id: conda_cache
-        with:
-          path: /tmp/test_env
-
-          key: ${{ runner.os }}-test-env-py38-\
-            ${{ hashFiles('tests/test-env-py38.yml') }}
 
-      - name: Update PATH
+      - name: Install Dependencies
         shell: bash
         run: |
-          echo "/tmp/test_env/bin" >> $GITHUB_PATH
+          apt update
+          DEBIAN_FRONTEND=noninteractive apt-get install -y \
+          --fix-missing --no-install-recommends \
+          nodejs \
+          zstd
+          python -m pip install --upgrade --no-cache pip
+          pip install --upgrade --no-cache -r \
+          tests/requirements.txt
```

Comment:
This line will update datacube-core version which could over time become different from the one used in the stats.

Reply:
it is meant to be. It's a regression test since we always release stats first and then update the docker image. Hence it's to test the results will remain the same with newer version of libs compared to the ones built in the current docker image. Then we'll be confident (sort of) that the new docker image will be working as expected. The same test will be integrated into the docker image build workflow as well.

Reply:
It makes sense, I didn't see the forward thinking.